examples/RNN_toy_example.py

```diff
@@ -0,0 +1,142 @@
+from blocks.bricks import Linear, Tanh, Softmax
+from blocks.bricks.cost import SquaredError
+from blocks.filter import VariableFilter
+from blocks.graph import ComputationGraph
+from blocks.bricks import WEIGHTS
+from blocks.initialization import IsotropicGaussian, Constant
+from blocks.datasets import Dataset, ContainerDataset
+from blocks.datasets.streams import DataStream
+from blocks.datasets.schemes import SequentialScheme
+from blocks.algorithms import (GradientDescent, Scale,
+                               StepClipping, CompositeRule)
+from blocks.extensions.monitoring import TrainingDataMonitoring
+from blocks.main_loop import MainLoop
+from blocks.extensions import FinishAfter, Printing
+from blocks.bricks.recurrent import SimpleRecurrent
+import matplotlib.pyplot as plt
+
+
+class ToyDataset(Dataset):
+    def __init__(self, data):
+        self.data = data
+
+    def get_data(self, state=None, request=None):
+        return self.data[request]
+
+
+def main(seq_u, seq_y, n_h, n_y, n_epochs):
+    # Building Model
+    u = T.tensor3('input_sequence')
+    input_to_state = Linear(name='input_to_state',
+                            input_dim=seq_u.shape[-1],
+                            output_dim=n_h)
+    u_transform = input_to_state.apply(u)
+    RNN = SimpleRecurrent(activation=Tanh(),
+                          dim=n_h, name="RNN")
+    h = RNN.apply(u_transform)  # h is hidden states in the RNN
+    state_to_output = Linear(name='state_to_output',
+                             input_dim=n_h,
+                             output_dim=seq_y.shape[-1])
+    y_hat = state_to_output.apply(h)
+    y_hat.name = 'output_sequence'
+
+    predict = theano.function(inputs=[u, ], outputs=y_hat)
+
+    # Cost
+    y = T.tensor3('target_sequence')
+    cost = SquaredError().apply(y, y_hat)
+    cost.name = 'MSE'
+
+    # Initialization
+    RNN.weights_init = state_to_output.weights_init = \
+        input_to_state.weights_init = IsotropicGaussian(0.01)
+    RNN.biases_init = state_to_output.biases_init = \
+        input_to_state.biases_init = Constant(0)
+    RNN.initialize()
+    state_to_output.initialize()
+    input_to_state.initialize()
+
+    # Data
+    dataset = ContainerDataset({'input_sequence': seq_u,
+                                'target_sequence': seq_y})
+    stream = DataStream(dataset)
+
+    # Training
+    algorithm = GradientDescent(cost=cost,
+                                step_rule=CompositeRule([StepClipping(10.0),
+                                                         Scale(0.01)]))
+    monitor = TrainingDataMonitoring([cost],
+                                     prefix="train",
+                                     after_every_epoch=True)
+    main_loop = MainLoop(data_stream=stream, algorithm=algorithm,
+                         extensions=[monitor,
+                                     FinishAfter(after_n_epochs=n_epochs),
+                                     Printing()])
+
+    main_loop.run()
```

Comment:
Okay, one minute is a bit too much for a test. I was just thinking what is the best way to test this script. I think we should just run for fewer epochs, so that it took not more than, let's say, 5 seconds.